espresso-tests/migration-test.bash

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env bash
+
+fail(){
+    echo "$*" 1>&2; exit 1;
+}
+
+set -euo pipefail
+set -a # automatically export all variables
+set -x # print each command before executing it, for debugging
+
+
+
+# Find directory of this script
+TEST_DIR="$(dirname $(readlink -f $0))"
+TESTNODE_DIR="$(dirname "$TEST_DIR")"
+ORBIT_ACTIONS_DIR="$TESTNODE_DIR/orbit-actions"
+
+cd "$TESTNODE_DIR"
+
+# Initialize a standard network not compatible with espresso
+./test-node.bash --simple --init-force --tokenbridge --detach
+
+# Start espresso sequencer node
+docker compose up espresso-dev-node --detach
+
+# Export environment variables in .env file
+. "$TEST_DIR/.env"
+
+# Overwrite the ROLLUP_ADDRESS for this test, it might not be the same as the one in the .env file
+ROLLUP_ADDRESS=$(docker compose run --entrypoint cat scripts /config/deployed_chain_info.json | jq -r '.[0].rollup.rollup' | tail -n 1 | tr -d '\r\n')
+
+# A convoluted way to get the address of the child chain upgrade executor, maybe there's a better way?
+INBOX_ADDRESS=$(docker compose run --entrypoint cat scripts /config/deployed_chain_info.json | jq -r '.[0].rollup.inbox' | tail -n 1 | tr -d '\r\n')
+L1_TOKEN_BRIDGE_CREATOR_ADDRESS=$(docker compose run --entrypoint cat scripts /tokenbridge-data/network.json | jq -r '.l1TokenBridgeCreator' | tail -n 1 | tr -d '\r\n')
+CHILD_CHAIN_UPGRADE_EXECUTOR_ADDRESS=$(cast call $L1_TOKEN_BRIDGE_CREATOR_ADDRESS 'inboxToL2Deployment(address)(address,address,address,address,address,address,address,address,address)' $INBOX_ADDRESS | tail -n 2 | head -n 1 | tr -d '\r\n')
+
+# Export l2 owner private key and address
+PRIVATE_KEY="$(docker compose run scripts print-private-key --account l2owner | tail -n 1 | tr -d '\r\n')"
+OWNER_ADDRESS="$(docker compose run scripts print-address --account l2owner | tail -n 1 | tr -d '\r\n')"
+
+
+cd "$ORBIT_ACTIONS_DIR"
+
+git submodule update --init
+yarn
```

Comment:
I think it's a bit weird to run these installation commands in the middle of the script. Let's move them to the start or out of this script.